core/src/main/scala/org/apache/spark/resource/ExecutorResourceRequests.scala

```diff
@@ -32,9 +34,9 @@ import org.apache.spark.resource.ResourceProfile._
  */
 private[spark] class ExecutorResourceRequests() extends Serializable {
 
-  private val _executorResources = new mutable.HashMap[String, ExecutorResourceRequest]()
+  private val _executorResources = new ConcurrentHashMap[String, ExecutorResourceRequest]()
```

Comment:
Is this change required due to this PR? Or, should we apply this change without this PR?

Reply:
it can apply without the executor side changes, but all of these classes are private right now so I just included it here.
I can split it off if you would like

Reply:
I am curious whether this concurrency issue also exists in the master and past branches.

Reply:
same answer as below for ResourceProfile